### ISSUE TITLE
Notifications not working.  MacOS Mojave

### ISSUE THREAD

**Issue by sasokess** [Foundry376/Mailspring#1467]
##### Are there any related issues?

I looked up, read, and tried every solution posted out there with no success.

...

##### What operating system are you using?

MacOS Mojave 10.14.3

##### What version of Mailspring are you using?

1.6.1

--

**Bug?**

The mailsync-**.log files contain very little PII and lots of debug info! -->

I am not receiving banner notifications of new emails at all.  However, notifications of read receipts and clicks are working fine, but not new emails!

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?

No.

##### Is the issue reproducible with a particular attachment, message, signature, etc?

Hard to tell.

--

**Comment by Armstrongzcl** [comment]
I meet the same problem on Ubuntu 18.04. Mailspring version is 1.6.1.Do you solve it?

**Comment by lalitjolania** [comment]
I have the same problem in Manjaro. After looking into system logs I found this --

Apr 18 13:40:25  mailspring.desktop[1480]: { TypeError: Cannot read property 'toLowerCase' of undefined
Apr 18 13:40:25  mailspring.desktop[1480]:     at __getAllIconPaths (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/linux-theme-u>
Apr 18 13:40:25  mailspring.desktop[1480]:     at getIconFromTheme (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/linux-theme-ut>
Apr 18 13:40:25  mailspring.desktop[1480]:     at getIconPath (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/linux-theme-utils.t>
Apr 18 13:40:25  mailspring.desktop[1480]:     at Object.getIcon (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/linux-theme-util>
Apr 18 13:40:25  mailspring.desktop[1480]:     at NativeNotifications.getIcon (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/nat>
Apr 18 13:40:25  mailspring.desktop[1480]:     at new NativeNotifications (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/native->
Apr 18 13:40:25  mailspring.desktop[1480]:     at Object.<anonymous> (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/native-notif>
Apr 18 13:40:25  mailspring.desktop[1480]:     at Object.<anonymous> (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/native-notif>
Apr 18 13:40:25  mailspring.desktop[1480]:     at Module._compile (internal/modules/cjs/loader.js:693:30)
Apr 18 13:40:25  mailspring.desktop[1480]:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
Apr 18 13:40:25  mailspring.desktop[1480]:     at Module.load (internal/modules/cjs/loader.js:602:32)
Apr 18 13:40:25  mailspring.desktop[1480]:     at tryModuleLoad (internal/modules/cjs/loader.js:541:12)
Apr 18 13:40:25  mailspring.desktop[1480]:     at Function.Module._load (internal/modules/cjs/loader.js:533:3)
Apr 18 13:40:25  mailspring.desktop[1480]:     at Module.require (internal/modules/cjs/loader.js:640:17)
Apr 18 13:40:25  mailspring.desktop[1480]:     at require (internal/modules/cjs/helpers.js:20:18)
Apr 18 13:40:25  mailspring.desktop[1480]:     at lazyLoadWithGetter (/usr/share/mailspring/resources/app.asar/src/global/mailspring-exports.js:31:54)
Apr 18 13:40:25  mailspring.desktop[1480]:     at Object.get [as NativeNotifications] (/usr/share/mailspring/resources/app.asar/src/global/mailspring-exports.js:23:21)
Apr 18 13:40:25  mailspring.desktop[1480]:     at Notifier._playNewMailSound.underscore_1.default.debounce (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-li>
Apr 18 13:40:25  mailspring.desktop[1480]:     at Notifier._playNewMailSound (/usr/share/mailspring/resources/app.asar/node_modules/underscore/underscore.js:841:23)
Apr 18 13:40:25  mailspring.desktop[1480]:     at mailspring_exports_1.DatabaseStore.findAll.then.threadsArray (file:///tmp/nylas-build/electron-packager/linux-x64/mailsprin>
Apr 18 13:40:25  mailspring.desktop[1480]:   message: 'Cannot read property \'toLowerCase\' of undefined' } { pluginIds: [ 'unread-notifications' ] }
Apr 18 13:40:25  mailspring.desktop[1480]: Raven: undefined - undefined

**Comment by ivnilv** [comment]
Happens to me as well ..

**Comment by aia832003** [comment]
Same on Fedora 29 and 30 beta with Gnome Shell.

**Comment by sasokess** [comment]
Are they going to solve this?
it's a huge killer!  I paid money to the app because I loved every other function.  But now I am keepin Mail open just for the notifications.
Please fix or respond at least!

**Comment by aia832003** [comment]
I've currently installed Mailnag as a workaround to get notifications, but really hope this gets fixed in next update.

**Comment by krupenja** [comment]
Can confirm that notifications on Mojave stopped working around a week ago, maybe after update to 1.6.1.

**Comment by FlamingFury017** [comment]
Notifications not working on Arch Linux with i3 too :/

**Comment by krupenja** [comment]
Had to temporarily switch to macOS Mail because of this. Hope this gets fixed soon.

**Comment by migalmoreno** [comment]
Also happening on Ubuntu 18.10 since I upgraded from v1.5.7, in which new email notifications were working fine.

**Comment by cyfrost** [comment]
Same situation, new mail notifications not working with Fedora 30 Workstation GNOME 3.32.1 and Zenity installed as required.

**Comment by robertoswald** [comment]
Same problem on elementary os 5.0, with mailspring 1.6.1. The weird thing is one time I saw a notification.

**Comment by bengotow** [comment]
Hey folks, it looks like the issue on Linux is due to some new code we added to support themeing the application icon (the `toLowerCase` problem.) Will get that fixed in the next release. On macOS we may need to do a bit more investigation!

**Comment by axfelix** [comment]
+1 to not working in Mojave. Let me know if you need logs or anything.

**Comment by axfelix** [comment]
They're working again today after several days. Maybe it couldn't hit some CDN? I love Electron :)

**Comment by wizardion** [comment]
It doesn’t work on Ubuntu 19.04, in my case it Unity, when the app is closed (in tray), notifications are not showing. When the app is active (opened window) - notifications are showing.

**Comment by MadByteDE** [comment]
> It doesn’t work on Ubuntu 19.04, in my case it Unity, when the app is closed (in tray), notifications are not showing. When the app is active (opened window) - notifications are showing.

Im on Ubuntu 19.04 as well & can confirm this behaviour. Hope this gonna be fixed soon™.

**Comment by polymoon** [comment]
I'm on Windows 10 Pro and notifications stopped working for me in 1.6.1 as well.

**Comment by timsabruno** [comment]
Just got the app and noticed notifications don't work. Mojave 10.14.5 Mailspring 1.6.1

**Comment by keith2519** [comment]
Yeah, notifications not working on Windows 10, Mailspring version 1.6.1
Please fix.

**Comment by keith2519** [comment]
> Hey folks, it looks like the issue on Linux is due to some new code we added to support themeing the application icon (the `toLowerCase` problem.) Will get that fixed in the next release. On macOS we may need to do a bit more investigation!

Hopefully Windows version will be fixed too in the next release.

**Comment by timsabruno** [comment]
> Just got the app and noticed notifications don't work. Mojave 10.14.5 Mailspring 1.6.1

Today, still version 1.6.1 on Mojave 10.14.5 notifications started working somehow

**Comment by 0x2141** [comment]
Hi, notifications are not working on Fedora 30 Gnome 3.32.2. It just stopped working.

**Comment by SebastianCApostolescu** [comment]
I'm on Windows 10 Pro x64 with Mailspring v. 1.6.2-278bf219 and notification works only sometimes. This has bees going on for a few months I think.

**Comment by robertoswald** [comment]
Same on elementary OS 5.0

**Comment by krupenja** [comment]
Can confirm this is still an issue with 1.6.2 on macos

**Comment by BjoernDaase** [comment]
> Hey folks, it looks like the issue on Linux is due to some new code we added to support themeing the application icon (the `toLowerCase` problem.) Will get that fixed in the next release. On macOS we may need to do a bit more investigation!

I'm on 1.6.2 on Linux and sadly notifications still don't work....

**Comment by al2me6** [comment]
Still an issue on 1.6.2, Manjaro Linux.

**Comment by drahoja9** [comment]
Notifications still not working on 1.6.3, Ubuntu 18.04

**Comment by Rogergonzalez21** [comment]
Still not working, 1.6.3, Manjaro Linux

**Comment by al2me6** [comment]
Pinging @bengotow,

This issue has persisted for almost four (!) months now and presents a significant impediment for those who are affected. Could you please provide an update on this issue? I'd say that it is unacceptable for a email client to be unable to notify the user of incoming emails.

**Comment by Rogergonzalez21** [comment]
I found out that on Manjaro, notifications came up when I powered up from a sleep, but stopped working right after. I don't know if this can help with the solution

**Comment by Armstrongzcl** [comment]
@bengotow when will the bug be fixed?

**Comment by Croccifixio** [comment]
For anyone that doesn't want to wait for a fix and is okay with running a slightly older build that supports desktop notifications, you can do the following:

1. Add this to your hosts file to disable auto updates:
```
127.0.0.1 updates.getmailspring.com
```

2. [Install version 1.6.0](https://github.com/Foundry376/Mailspring/releases/tag/1.6.0) of Mailspring (at least on Windows this seems like like the last  that supported desktop notifications).

**Comment by aia832003** [comment]
Any update from the Mailspring team on this? We are entering month 5 of this issue @bengotow ?

**Comment by Oleksandr1** [comment]
mailspring 1.6.3
macos 10.14.6 
The problem is still be actual.

**Comment by jbsledzik** [comment]
Ubuntu 18.04, the problem also still exists. Temporary, I restored `1.6.0` version according to @Croccifixio comment, but I hope will be fixed soon.

**Comment by naclcaleb** [comment]
Problem still exists on macOS

**Comment by kolexander** [comment]
I reinstalled to version 1.6.0 and still don't see the number of unread emails in the tray ...

**Comment by afxgak** [comment]
> Problem still exists on macOS

Hi everyone, I had the same problem, but somehow I solved the problem.
I made fresh install (new user here) and faced the same problem.
Here what I've done: 
open application folder on mac
open system preferences - notifications 
drag and drop mailspring app icon to the list of apps in notification panel
mailspring should be open, but in background
send email to test
close system prefs and open them again, check for mailspring in the list of apps

I've tried this twice before register to write here. Now its working fine.

**Comment by bengotow** [comment]
Hey folks! Sorry for the delay on this one—it looks like there were a few separate things going on here that made it tricky to debug. Mailspring only displays notifications for "new" mail, but internally it was basically saying "new == just saved to the database." A performance change to the mailsync codebase has been causing messages to be created and then updated with their full message bodies and headers a lot faster, and if the JavaScript / Electron process falls behind by the time it sees the message it's no longer "new." But whether that happens depends on the speed of your computer, the volume of email you're getting all at one time, etc.

Long story short, the fix is in d561d7d6 and is fairly simple. The sync engine explicitly tags messages as "new" so that the speed at which they're seen doesn't matter, and manual tests on Linux and Windows indicate things are working fine again. Sorry for the hassle, this will ship this week!

**Comment by axfelix** [comment]
Hooray for race conditions!!!

**Comment by aia832003** [comment]
@bengotow will this still ship in a release this week?

**Comment by mari5ha** [comment]
MX linux 18.3 - mailspring 1.6.3. still have no notification issue. I install zenity already but there are no new mail notification

**Comment by krupenja** [comment]
Installed 1.7.0 on macOS Catalina: now i am getting the notification *sound* but not getting the actual notification *popup* when I receive e-mail.

**Comment by krupenja** [comment]
Same thing with 1.7.1.

**Comment by Armstrongzcl** [comment]
It works, thanks!

**Comment by axfelix** [comment]
Good here too! @krupenja , any chance that's just your macOS notification settings? I think this can be closed.

**Comment by krupenja** [comment]
@krupenja Yeah, I as also thinking about it being an issue with some new crap Catalina settings. The thing is, when I open Notification settings I do not see Mailspring there at all (other apps are present). Any ideas?

**Comment by krupenja** [comment]
OK, finally got notifications to work. Had to do the following steps:

1. Delete `~/Library/Preferences/com.apple.ncprefs.plist` file.
2. Reboot.
3. Open Mailspring and wait for an e-mail to arrive (or send yourself an e-mail).
4. Notification asking for permissions appears, click Show.

**Comment by bengotow** [comment]
Hey folks! Thanks everyone for reporting this and for being so patient as we ship a fix. The 1.7.x releases address this issue and notifications should be working properly now. 

It sounds like there may be something going on specific to macOS Catalina and notification permissions. If you hear the new mail sound but don't see a notification, please file another issue and we'll track that one there. Reading the comments it sounds like this may have been broken and then fixed in the Catalina betas, so also double check that you're running the latest macOS.

Thanks! You can download the update from our website at getmailspring.com/download or choose "Check for Updates" in the app's menu.
